packages/@sanity/vision/src/components/QueryEditor.js

```diff
@@ -2,13 +2,43 @@
 /* eslint-disable import/no-unassigned-import */
 import React from 'react'
 import PropTypes from 'prop-types'
-import {UnControlled as ReactCodeMirror} from 'react-codemirror2'
+import {UnControlled as BaseReactCodeMirror} from 'react-codemirror2'
 import CodeMirror from 'codemirror'
+import styled from 'styled-components'
+import {hues} from '@sanity/color'
+import {rem} from '@sanity/ui'
 
 require('codemirror/mode/javascript/javascript')
 require('codemirror/addon/hint/show-hint')
 require('codemirror/addon/edit/closebrackets')
 
+const ReactCodeMirror = styled(BaseReactCodeMirror)`
+  width: 100%;
+  box-sizing: border-box;
+  height: 100%;
+  overflow: hidden;
+  position: relative;
+
+  .CodeMirror {
+    width: 100%;
+    height: 100%;
+    overflow: hidden;
+    position: relative;
+  }
+
+  .CodeMirror-sizer {
+    padding-top: ${({theme}) => rem(theme.sanity.space[5])};
+  }
+
+  .CodeMirror-linenumber {
+    padding: 0 3px 0 5px;
+    min-width: 1.5rem;
+    text-align: right;
+    color: ${hues.gray['600'].hex};
```

Comment:
In order for this to work with the custom theming people might have in the Studio, we need to use the CSS variables:
```suggestion
    color: var(--card-code-fg-color);
```